src/protocol/context/malicious.rs

```diff
@@ -741,19 +669,15 @@ where
     F: Field,
 {
     async fn upgrade(self, input: Replicated<F>) -> Result<MaliciousReplicated<F>, Error> {
-        self.inner
-            .upgrade_one(
-                self.upgrade_ctx,
-                self.record_binding,
-                input,
-                ZeroPositions::Pvvv,
-            )
+        self.ctx
+            .upgrade_one(self.record_binding, input, ZeroPositions::Pvvv)
             .await
     }
 }
 
 // Impl for upgrading things that can be upgraded using a single record ID using a non-record-bound
-// context. This gets used e.g. when the protocol takes a single `Replicated<F>` input.
+// context. This is only used for tests where the protocol takes a single `Replicated<F>` input.
+#[cfg(test)]
```

Comment:
These functions were test-only, which is good, because they are a bit fragile.  Some tests now set total records (to 1), but others don't.  Rather than try to standardize, this was easier.